src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/action/start/TransportStartTransformAction.kt

```diff
@@ -0,0 +1,150 @@
+                if (transform == null) {
+                    return actionListener.onFailure(
+                        ElasticsearchStatusException("Could not find transform [${request.id()}]",
+                            RestStatus.NOT_FOUND)
+                    )
+                }
+
+                if (transform.enabled) {
+                    log.debug("Transform job is already enabled, checking if metadata needs to be updated")
+                    return if (transform.metadataId == null) {
+                        actionListener.onResponse(AcknowledgedResponse(true))
+                    } else {
+                        getTransformMetadata(transform, actionListener)
+                    }
+                }
+
+                updateTransformJob(transform, request, actionListener)
+            }
+
+            override fun onFailure(e: Exception) {
+                actionListener.onFailure(ExceptionsHelper.unwrapCause(e) as Exception)
+            }
+        })
+    }
+
+    private fun updateTransformJob(
+        transform: Transform,
+        request: StartTransformRequest,
+        actionListener: ActionListener<AcknowledgedResponse>
+    ) {
+        val now = Instant.now().toEpochMilli()
+        request.index(INDEX_MANAGEMENT_INDEX).doc(mapOf(Transform.TRANSFORM_TYPE to mapOf(Transform.ENABLED_FIELD to true,
+            Transform.ENABLED_AT_FIELD to now, Transform.UPDATED_AT_FIELD to now)))
+        client.update(request, object : ActionListener<UpdateResponse> {
+            override fun onResponse(response: UpdateResponse) {
+                if (response.result == DocWriteResponse.Result.UPDATED) {
+                    // If there is a metadata ID on transform then we need to set it back to STARTED or RETRY
+                    if (transform.metadataId != null) {
+                        getTransformMetadata(transform, actionListener)
+                    } else {
+                        actionListener.onResponse(AcknowledgedResponse(true))
+                    }
+                } else {
+                    actionListener.onResponse(AcknowledgedResponse(false))
+                }
+            }
+
+            override fun onFailure(e: Exception) {
+                actionListener.onFailure(ExceptionsHelper.unwrapCause(e) as Exception)
+            }
+        })
+    }
+
+    private fun getTransformMetadata(transform: Transform, actionListener: ActionListener<AcknowledgedResponse>) {
+        val req = GetRequest(INDEX_MANAGEMENT_INDEX, transform.metadataId).routing(transform.id)
+        client.get(req, object : ActionListener<GetResponse> {
+            override fun onResponse(response: GetResponse) {
+                if (!response.isExists || response.isSourceEmpty) {
+                    // If there is no metadata doc then the runner will instantiate a new one
+                    // in FAILED status which the user will need to retry from
+                    actionListener.onResponse(AcknowledgedResponse(true))
```

Comment:
Just thinking through this, during transform run we always first create transformMetadata and then associate to the transform document. So if an metadata id exists in transform but there is document then the only reason will be someone delete the transform metadata manually. I think we should just throw an exception here metadata is in weird state and cannot be started